### PR TITLE
bump zustand + add performance measurements

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Commit Details:**
- Bumping Zustand, and fixing the breaking API changes
- Added a new way to performance.measure the zustand selectors by wrapping them in a utility function

### CODE REVIEW

File: editor/src/components/editor/store/editor-state.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ export const defaultUserState: UserState = {
   shortcutConfig: {},
 }
 
-export interface EditorStore {
+export type EditorStore = {
```
Comment (Rheeseyb) [inline editor/src/components/editor/store/editor-state.ts:159]: Why the change from `interface` to `type` here?

**Review by Rheeseyb — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

File: editor/src/components/editor/store/editor-state.ts (hunk 1 above)
Comment (balazsbajorics) [reply editor/src/components/editor/store/editor-state.ts:159]: This is a bit unclear to me.

There was a compile error that went like this:
```
Type 'EditorStore' does not satisfy the constraint 'Record<string | number | symbol, unknown>'.
  Index signature is missing in type 'EditorStore'.ts
```
Changing EditorStore from interface to type fixed this error. I don't quite know why!
The change itself means EditorStore is now a type alias to the `typeof` of the object literal type. 
Maybe that means it also gets some object-like features like an index signature?
